temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/RootNamespaceAutoConfiguration.java

```diff
@@ -149,15 +150,23 @@ public WorkerFactoryStarter workerFactoryStarter(WorkerFactory workerFactory) {
     return new WorkerFactoryStarter(workerFactory);
   }
 
-  public static class WorkerFactoryStarter implements ApplicationListener<ContextRefreshedEvent> {
+  // It needs to listed on ContextStartedEvent, not ContextRefreshedEvent.
+  // Using ContextRefreshedEvent will cause start of workers early, during the context
+  // initialization
+  // and potentially incorrect order of bean initialization.
+  // Refresh event can also be fired multiple times during the context initialization.
+  // For this listener to ever work, Soring context needs to be actually started.
+  // Note that a lot of online samples for Spring don't start the context at all:
+  // https://stackoverflow.com/questions/48099355/contextstartedevent-not-firing-in-custom-listener
```

Comment:
I wonder if it's worth having a test that would break if we hadn't changed this code (i.e. a test that would fail if refreshed during initialization or context not started or whatever).

Reply:
Added a separate test verifying the influence of initialization/refresh and start on WorkerFactory.